frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/GoService.java

```diff
@@ -0,0 +1,104 @@
+package ch.shiftcrypto.bitboxapp;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Intent;
+import android.graphics.drawable.Icon;
+import android.os.Binder;
+import android.os.Build;
+import android.os.IBinder;
+
+import androidx.core.app.NotificationCompat;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import goserver.GoAPIInterface;
+import goserver.GoEnvironmentInterface;
+import goserver.Goserver;
+
+public class GoService extends Service {
+    // Binder given to clients
+    private final IBinder binder = new GoServiceBinder();
+
+    private boolean started = false;
+
+    private Lock startedLock = new ReentrantLock();
+
+    private final String channelId = "21";
+
+    private final int notificationId = 8;
+
+    @Override
+    public void onCreate() {
+        Util.log("GoService onCreate()");
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel channel = new NotificationChannel(
+                    channelId,
+                    "BitBoxApp",
+                    NotificationManager.IMPORTANCE_LOW
+            );
+            channel.setDescription("BitBoxApp service notification channel");
+            NotificationManager manager = getSystemService(NotificationManager.class);
+            manager.createNotificationChannel(channel);
+        }
+        // Create a PendingIntent that starts the main activity of your app
+        Intent notificationIntent = new Intent(getApplicationContext(),MainActivity.class);
+        notificationIntent.setAction(Intent.ACTION_MAIN);
+        notificationIntent.addCategory(Intent.CATEGORY_LAUNCHER);
+        notificationIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+        PendingIntent contentIntent = PendingIntent.getActivity(
+                getApplicationContext(),
+                0,
+                notificationIntent,
+                PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
Adding `| PendingIntent.FLAG_IMMUTABLE` here seems to fix the crash I commented about.

Reply:
Solution courtesy of ChatGPT! 

![](https://user-images.githubusercontent.com/1225267/222759380-3ea7d5d7-5710-4254-90e5-b1dd68bc4add.png)

Reply:
Great! I pushed a new commit that should fix :) Thanks